### DISCUSSION TITLE
A way to reset a component

### DISCUSSION THREAD

**Discussion by timrspratt** [livewire/livewire#1552] [4. General]
Would it be possible to add a `$reset` magic action? The use case is resetting a component to its original state. Say you have a `NewUser` component, that is a form to enter the information for a new user (dur!), and when saving the user, the form can reset (i.e. remove the property values and empty the form input fields).

**Comment by timrspratt** [comment]
The problem with `$this->reset()` on the PHP side, is that it only offers a way to reset all or specific ones and doesn't fire the `mount` method

**Comment by imliam** [comment]
I'm not sure how possible it'd be to rerun the `mount` method after initial page load - the `mount` method can accept any kind of PHP object as a parameter in the initial page load, but subsequent renders are done from converting the object to something that can live in JS (hence why only scalars/models/collections/datetimes are supported, and not everything).

I guess I don't understand what `$this->reset()` doesn't do that you need it to. Why do you need to rerun the mount method? If you need to do more than _just_ resetting props (what other state do you even need to change?) why can't you call `$this->reset(); $this->setUpNewForm();` and have a 2nd method do the same as mount?

**Reply by timrspratt** [reply]
I'm currently doing just that @imliam, calling `$this->reset();` and then another method that is also called in `mount`. Essentially one of the properties is set in mount, which `reset` is removing. I wonder therefore if there is scope for a `$this->resetExcept(...$parameters)`, where it's a "reset all apart from these"?
